pyiceberg/catalog/rest/__init__.py

```diff
@@ -843,6 +998,13 @@ def table_exists(self, identifier: str | Identifier) -> bool:
         Returns:
             bool: True if the table exists, False otherwise.
         """
+        if Capability.V1_TABLE_EXISTS not in self._supported_endpoints:
```

Comment:
same as above, add a note about this special case

https://github.com/apache/iceberg/blob/0069c5e09617d99a3ddce33a4a093fe288243eeb/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L375-L392

https://github.com/apache/iceberg/blob/0069c5e09617d99a3ddce33a4a093fe288243eeb/api/src/main/java/org/apache/iceberg/catalog/SessionCatalog.java#L180-L187